internal/api/v1/stage.go

```diff
@@ -235,71 +234,3 @@ func createProductionCertificate(ctx context.Context, cfg *rest.Config, app gite
 
 	return nil
 }
-
-func createLocalCertificate(ctx context.Context, cfg *rest.Config, app gitea.App, systemDomain string) error {
-	data := fmt.Sprintf(`{
-		"apiVersion": "quarks.cloudfoundry.org/v1alpha1",
-		"kind": "QuarksSecret",
-		"metadata": {
-			"name": "%s",
-			"namespace": "%s"
-		},
-		"spec": {
-			"request" : {
-				"certificate" : {
-					"CAKeyRef" : {
-						"key" : "private_key",
-						"name" : "ca-cert"
-					},
-					"CARef" : {
-						"key" : "certificate",
-						"name" : "ca-cert"
-					},
-					"commonName" : "%s.%s",
-					"isCA" : false,
-					"alternativeNames": [
-						"%s.%s"
-					],
-					"signerType" : "local"
-				}
-			},
-			"secretName" : "%s-tls",
```

Comment:
The application ingress [is still expecting a secret named like this](https://github.com/epinio/epinio/blob/main/internal/cli/clients/gitea/upload.go#L312) (e.g. `sample-tls`) but the new code generates a secret named something like `sample-6f2kd`. This creates an ingress that doesn't work. Visiting the app route with `https` results in a 404.

Reply:
Turns out the wrong name was due to the typo above (https://github.com/epinio/epinio/pull/385/files#r631744787). I fixed it and now I get the correct `*-tls` name for the generated secret. cert-manager seems to handle the wrong issuer name with some fallback (?). I wonder why no test failed. We should probably add a test that would fail if this happens again.

Reply:
I added a test and fixed the issue